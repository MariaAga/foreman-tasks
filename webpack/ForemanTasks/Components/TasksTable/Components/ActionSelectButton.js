import React from 'react';
import { DropdownButton, MenuItem } from 'patternfly-react';
import PropTypes from 'prop-types';
import { translate as __ } from 'foremanReact/common/I18n';

export const ActionSelectButton = ({ onCancel, onResume, disabled }) => (
  <DropdownButton
    title={__('Select Action')}
    disabled={disabled}
    id="selcted-action-type"
  >
    <MenuItem
      title={__('Cancel selected tasks')}
      onClick={onCancel}
      eventKey="1"
    >
      {__('Cancel Selected')}
    </MenuItem>
    <MenuItem
      title={__('Resume selected tasks')}
      onClick={onResume}
      eventKey="2"
    >
      {__('Resume Selected')}
    </MenuItem>
  </DropdownButton>
);

ActionSelectButton.propTypes = {
  disabled: PropTypes.bool,
  onCancel: PropTypes.func.isRequired,
  onResume: PropTypes.func.isRequired,
};

ActionSelectButton.defaultProps = {
  disabled: false,
};
