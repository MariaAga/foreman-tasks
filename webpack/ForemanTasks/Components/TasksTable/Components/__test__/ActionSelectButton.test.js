import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import { ActionSelectButton } from '../ActionSelectButton';

const fixtures = {
  'renders with minimal props': {
    onCancel: jest.fn(),
    onResume: jest.fn(),
  },
};

describe('ActionSelectButton', () =>
  testComponentSnapshotsWithFixtures(ActionSelectButton, fixtures));
