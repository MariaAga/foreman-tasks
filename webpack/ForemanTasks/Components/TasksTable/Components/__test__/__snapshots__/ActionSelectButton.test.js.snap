// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionSelectButton renders with minimal props 1`] = `
<DropdownButton
  disabled={false}
  id="selcted-action-type"
  title="Select Action"
>
  <MenuItem
    bsClass="dropdown"
    disabled={false}
    divider={false}
    eventKey="1"
    header={false}
    onClick={[MockFunction]}
    title="Cancel selected tasks"
  >
    Cancel Selected
  </MenuItem>
  <MenuItem
    bsClass="dropdown"
    disabled={false}
    divider={false}
    eventKey="2"
    header={false}
    onClick={[MockFunction]}
    title="Resume selected tasks"
  >
    Resume Selected
  </MenuItem>
</DropdownButton>
`;
